pkgs/os-specific/linux/trelay/default.nix

```diff
@@ -0,0 +1,47 @@
+{ stdenv, lib, fetchFromGitHub, kernel, kmod }:
+let
+  version = "22.03.5";
+in
+stdenv.mkDerivation {
+  name = "trelay";
+  version = "${version}-${kernel.version}";
+
+  src = fetchFromGitHub {
+    owner = "openwrt";
+    repo = "openwrt";
+    rev = "v${version}";
+    hash = "sha256-Q0EdFPlOQl4XUtifAupNPor/QRthtijo7Xgo7rikW84=";
+    sparseCheckout = [ "package/kernel/trelay" ];
+  };
+
+  sourceRoot = "source/package/kernel/trelay/src";
+  hardeningDisable = [ "pic" "format" ];
+  nativeBuildInputs = [ kmod ] ++ kernel.moduleBuildDependencies;
+  broken = [ linuxPackages_hardkernel_latest.trelay ];
+
+  postPatch = ''
+    cp '${./Makefile}' Makefile
+  '';
+
+  makeFlags = [
+    "KERNELRELEASE=${kernel.modDirVersion}"
+    "KERNEL_DIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "INSTALL_MOD_PATH=$(out)"
+  ];
+
+  meta = with lib; {
+    description = "Trelay relays IP packets between two devices to build a IP bridge between them.";
```

Comment:
description is not supposed to start with (or mention) the package name, and should not contain punctuation